src/rpdk/core/contract/resource_client.py

```diff
@@ -69,6 +69,30 @@ def override_properties(document, overrides):
     return document
 
 
+def prune_properties_except_identifier(model, paths):
+    """Prune all properties except given ones from the document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    document = {"properties": model}
+    for path in list(model):
+        path_tmp = (
+            "properties",
+            path,
+        )
+        _prop, resolved_path, parent = traverse(document, path_tmp)
+        key = resolved_path[-1]
+        key_path = (
+            "properties",
+            key,
+        )
+        if key_path not in paths:
+            del parent[key]
+    return document["properties"]
```

Comment:
why all these needed? u have access to schema; schema is a map which gives u primary identifier key with O(1) as it's not nested deep (
```json
{
    "properties": {
        ...
        "primaryIdentifier": [...],
        ...
    }
}
```
). from model u can get value with O(1) and return `return {"properties": {"id": "value"}}`; seems that this method can be two lines of code.